accounts-db/src/accounts.rs

```diff
@@ -44,8 +44,25 @@ use {
     },
 };
 
+// max number of accts an entry can hold: 64 * 256
+const ENTRY_ACCTS_LOOKUP_TABLE_SIZE: usize = 16384;
 pub type PubkeyAccountSlot = (Pubkey, AccountSharedData, Slot);
 
+#[derive(Debug, Default)]
+struct EntryAcctLocks {
+    writables: HashSet<Pubkey>,
+    readables: HashSet<Pubkey>,
+}
+
+impl EntryAcctLocks {
+    fn with_capacity(capacity: usize) -> Self {
+        EntryAcctLocks {
+            writables: HashSet::with_capacity(capacity),
+            readables: HashSet::with_capacity(capacity),
+        }
+    }
+}
+
```

Comment:
below in `AccountLocks`, I'm considering we may want to have `write_locks` also track the number of outstanding write-locks, even if all must be in same batch.

It makes it so that the sanity checks we do when unlocking batches not broken by the new functionality.